service/api/types.go

```diff
@@ -414,6 +414,9 @@ const (
 	// SwitchGoroutine switches the debugger's current thread context to the thread running the specified goroutine
 	SwitchGoroutine = "switchGoroutine"
 	// Halt suspends the process.
+	// The effect of Halt while the target process is stopped, or in the
+	// process of stopping, is operating system and timing dependent. It will
+	// either have no effect or cause the following resume to stop immediately.
```

Comment:
In that case, the dap pause should be checking is-running and not calling halt if not running, shouldn't it?

Reply:
Maybe. Doesn't the stop event disable the pause button anyway? The DAP specification doesn't say what the pause request does when the target is stopped anyway.